### ISSUE TITLE
IndexError when calculating median for `--filter cn`

### ISSUE THREAD

**Issue by chapmanb** [etal/cnvkit#217]
Eric;
I'd recently added `--filter cn` into bcbio to help merge adjacent calls with identical copy numbers. Thanks again for your tip on that. We're running into a few issues in cases where we end up with an empty array when calculating the median:
```
Calling copy number with thresholds: -1.1 => 0, -0.25 => 1, 0.2 => 2, 0.7 => 3
Applying filter 'cn'
Traceback (most recent call last):
  File "/g/data/gx8/local/share/bcbio/anaconda/bin/cnvkit.py", line 13, in <module>
    args.func(args)
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/cnvlib/commands.py", line 685, in _cmd_call
    args.thresholds)
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/cnvlib/call.py", line 71, in do_call
    outarr = getattr(segfilters, filt)(outarr)
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/cnvlib/segfilters.py", line 30, in wrapped_f
    result = func(segarr)
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/cnvlib/segfilters.py", line 137, in cn
    return squash_by_groups(segarr, segarr['cn'])
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/cnvlib/segfilters.py", line 53, in squash_by_groups
    data = data.groupby(groupkey, sort=False).apply(squash_region)
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/pandas/core/groupby.py", line 694, in apply
    return self._python_apply_general(f)
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/pandas/core/groupby.py", line 698, in _python_apply_general
    self.axis)
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/pandas/core/groupby.py", line 1601, in apply
    res = f(group)
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/cnvlib/segfilters.py", line 86, in squash_region
    out['cn1'] = weighted_median(cnarr['cn1'], cnarr['weight'])
  File "/g/data/gx8/local/share/bcbio/anaconda/lib/python2.7/site-packages/cnvlib/descriptives.py", line 124, in weighted_median
    return a[midpoint_idx]
IndexError: index 0 is out of bounds for axis 0 with size 0
' returned non-zero exit status 1
```
I'd be happy to skip the step in cases where it wouldn't work but wasn't able to come up with a good check in the case where it's failing. The original array is not empty so checking on `len(cnarr['cn1'])` doesn't catch it. Do you have any recommendations for debugging/avoiding this? Thanks much for any suggestions.

**Comment by etal** [comment]
I might have fixed this in the current development version. I'll take a look.

**Comment by etal** [comment]
Here's an earlier report of the bug: etal/cnvkit#202, fixed in commit 101d4a660f77d302e7557b6361bcc5d8fe23196c

The current development version will become 0.9.0. I'm not aware of anything that would make it unsuitable to use in bcbio now, but it does remove some deprecated functionality and split out another top-level namespace, `skgenome`, alongside `cnvlib`. It's compatible with the new pandas version 0.20, too.

**Comment by chapmanb** [comment]
Eric -- apologies for missing the previous report and thanks for the pointer. I'll update the bioconda package to the latest development and re-enable filtering in bcbio after that's in place. Sorry again for the noise and thanks much.
